Nodejs/Product/Nodejs/SharedProject/CommonPropertyPage.cs

```diff
@@ -1,342 +1,27 @@
-        }
-
-        public bool Loading { get; set; }
-
-        public bool IsDirty
-        {
-            get
-            {
-                return this.dirty;
-            }
-            set
-            {
-                if (this.dirty != value && !this.Loading)
-                {
-                    this.dirty = value;
-                    if (this.site != null)
-                    {
-                        this.site.OnStatusChange((uint)(this.dirty ? PropPageStatus.Dirty : PropPageStatus.Clean));
-                    }
-                }
-            }
-        }
-
-        void IPropertyPage.Activate(IntPtr hWndParent, RECT[] pRect, int bModal)
-        {
-            this.Control.Visible = false;
-
-            // suspend to reduce flashing
-            this.Control.SuspendLayout();
-
-            try
-            {
-                var parent = Control.FromHandle(hWndParent);
-                this.Control.Parent = parent;
-
-                // move to final location
-                ((IPropertyPage)this).Move(pRect);
-            }
-            finally
-            {
-                this.Control.ResumeLayout();
-                this.Control.Visible = true;
-                this.Control.Focus();
-            }
-        }
-
-        int IPropertyPage.Apply()
-        {
-            try
-            {
-                Apply();
-                return VSConstants.S_OK;
-            }
-            catch (Exception e)
-            {
-                return Marshal.GetHRForException(e);
-            }
-        }
-
-        void IPropertyPage.Deactivate()
-        {
-            this.Project = null;
-            this.Control.Dispose();
-        }
-
-        void IPropertyPage.GetPageInfo(PROPPAGEINFO[] pPageInfo)
-        {
-            Utilities.ArgumentNotNull("pPageInfo", pPageInfo);
-
-            var info = new PROPPAGEINFO();
-
-            info.cb = (uint)Marshal.SizeOf(typeof(PROPPAGEINFO));
-            info.dwHelpContext = 0;
-            info.pszDocString = null;
-            info.pszHelpFile = null;
-            info.pszTitle = this.Name;
-            info.SIZE.cx = this.DefaultSize.Width;
-            info.SIZE.cy = this.DefaultSize.Height;
-            pPageInfo[0] = info;
-        }
-
-        void IPropertyPage.Help(string pszHelpDir)
-        {
-            // not implemented
-        }
-
-        int IPropertyPage.IsPageDirty()
-        {
-            return (this.IsDirty ? (int)VSConstants.S_OK : (int)VSConstants.S_FALSE);
-        }
-
-        void IPropertyPage.Move(RECT[] pRect)
-        {
-            Utilities.ArgumentNotNull("pRect", pRect);
-
-            var r = pRect[0];
-
-            this.Control.Location = new Point(r.left, r.top);
-            this.Control.Size = new Size(r.right - r.left, r.bottom - r.top);
-        }
-
-        void IPropertyPage.SetObjects(uint count, object[] punk)
+        protected override void SetObjects(uint count, object[] punk)
```

Comment:
Can we do better than "punk"?

Reply:
they're either configs or properties... so not really.